cpp/src/barretenberg/dsl/acir_format/recursion_constraint.cpp

```diff
@@ -0,0 +1,356 @@
+        }
+    }
+
+    verification_key_data vkey_data{
+        .composer_type = vkey->composer_type,
+        .circuit_size = static_cast<uint32_t>(vkey->circuit_size),
+        .num_public_inputs = static_cast<uint32_t>(vkey->num_public_inputs),
+        .commitments = vkey->commitments,
+        .contains_recursive_proof = vkey->contains_recursive_proof,
+        .recursive_proof_public_input_indices = vkey->recursive_proof_public_input_indices,
+    };
+    output.emplace_back(vkey_data.compress_native(0)); // key_hash
+    return output;
+}
+
+/**
+ * @brief When recursively verifying proofs, we represent the verification key using field elements.
+ *        This method exports the key formatted in the manner our recursive verifier expects.
+ *        A dummy key is used when building a circuit without a valid witness assignment.
+ *        We want the transcript to contain valid G1 points to prevent on-curve errors being thrown.
+ *        We want a non-zero circuit size as this element will be inverted by the circuit
+ *        and we do not want an "inverting 0" error thrown
+ *
+ * @return std::vector<barretenberg::fr>
+ */
+std::vector<barretenberg::fr> export_dummy_key_in_recursion_format(const PolynomialManifest& polynomial_manifest,
+                                                                   const bool contains_recursive_proof)
+{
+    std::vector<fr> output;
+    output.emplace_back(1); // domain.domain (will be inverted)
+    output.emplace_back(1); // domain.root (will be inverted)
+    output.emplace_back(1); // domain.generator (will be inverted)
+
+    output.emplace_back(1); // circuit size
+    output.emplace_back(1); // num public inputs
+
+    output.emplace_back(contains_recursive_proof); // contains_recursive_proof
+    for (size_t i = 0; i < 16; ++i) {
+        output.emplace_back(0); // recursive_proof_public_input_indices
+    }
+
+    for (const auto& descriptor : polynomial_manifest.get()) {
+        if (descriptor.source == PolynomialSource::SELECTOR || descriptor.source == PolynomialSource::PERMUTATION) {
+            // the std::biggroup class creates unsatisfiable constraints when identical points are added/subtracted.
+            // (when verifying zk proofs this is acceptable as we make sure verification key points are not identical.
+            // And prover points should contain randomness for an honest Prover).
+            // This check can also trigger a runtime error due to causing 0 to be inverted.
+            // When creating dummy verification key points we must be mindful of the above and make sure that each
+            // transcript point is unique.
+            auto scalar = barretenberg::fr::random_element();
+            const auto element = barretenberg::g1::affine_element(barretenberg::g1::one * scalar);
+            const uint256_t x = element.x;
+            const uint256_t y = element.y;
+            const barretenberg::fr x_lo = x.slice(0, 136);
```

Comment:
Could we replace these magic numbers with derived values? These come from `proof_system::plonk::NUM_LIMB_BITS_IN_FIELD_SIMULATION` (68)

Reply:
Made a new function `export_g1_affine_element_as_fields` that uses constexprs to determine the limb slices